Source/JavaScriptCore/jsc.cpp

```diff
@@ -916,6 +917,24 @@ JSInternalPromise* GlobalObject::moduleLoaderImportModule(JSGlobalObject* global
     if (!moduleURL.isLocalFile())
         RELEASE_AND_RETURN(scope, rejectWithError(createError(globalObject, makeString("Module url, '", moduleURL.string(), "' does not map to a local file."))));
 
+    auto assertions = JSC::retrieveAssertionsFromDynamicImportOptions(globalObject, parameters, { vm.propertyNames->type.impl() });
+    RETURN_IF_EXCEPTION(scope, promise->rejectWithCaughtException(globalObject, scope));
+
+    parameters = jsUndefined();
+    if (!assertions.isEmpty()) {
+        ScriptFetchParameters::Type type = ScriptFetchParameters::Type::None;
+        auto typeString = assertions.get(vm.propertyNames->type.impl());
+        if (typeString == "json"_s)
```

Comment:
Can we have a helper function which converts a type string to `ScriptFetchParameters::Type`?

Reply:
Yeah, added `ScriptFetchParameters::parseType` :)